src/sentry/api/serializers/models/alert_rule.py

```diff
@@ -330,20 +333,33 @@ def get_attrs(
                 incident_map[incident.id] = serialize(incident, user=user)
 
         serialized_alert_rules = serialize(alert_rules, user=user)
-        rules = serialize(
+        serialized_map_by_id = {x.id: x for x in serialized_alert_rules}
+
+        serialized_issue_rules = serialize(
             [x for x in item_list if isinstance(x, Rule)],
             user=user,
             serializer=RuleSerializer(expand=self.expand),
         )
+        serialized_issue_rule_map_by_id = {x.id for x in serialized_issue_rules}
 
         for item in item_list:
-            if isinstance(item, AlertRule):
-                alert_rule = serialized_alert_rules.pop(0)
-                if "latestIncident" in self.expand and alert_rule:
-                    alert_rule["latestIncident"] = incident_map.get(item.incident_id)  # type: ignore[attr-defined]
-                results[item] = alert_rule
-            elif isinstance(item, Rule):
-                results[item] = rules.pop(0)
+            item_id = str(item.id)
+            if item_id in serialized_map_by_id:
+                serialized_alert_rule = serialized_map_by_id[item_id]
+                if "latestIncident" in self.expand:
+                    serialized_alert_rule["latestIncident"] = incident_map.get(item.incident_id)  # type: ignore[attr-defined]
+                results[item] = serialized_alert_rule
+            elif item_id in serialized_issue_rule_map_by_id:
+                results[item] = serialized_issue_rule_map_by_id[item_id]
+            else:
+                logger.warning(
+                    "Alert Rule found but dropped during serialization",
```

Comment:
is there a case when a user creates an alert rule and it fails serialization, but it is created in the db? i'm a little nervous about if a user has an alert rule that can be triggered, but isn't visible via the API. If that's possible maybe we should ramp this up to a logger.error?

Reply:
Yea - i'll make this an error instead 👍